qutip/tests/conftest.py

```diff
@@ -32,6 +32,10 @@ def _skip_cython_tests_if_unavailable(item):
         # importorskip does not have maxversion
         if _version2int(_Cython.__version__) >= _version2int("3.0.0"):
             pytest.skip("cython 3.0.0 not supported")
+            return
+        import qutip
+        if not qutip.settings.has_cython:
+            pytest.skip("Cython not available at runtime.")
```

Comment:
I'm a bit uncomfortable relying on QuTiP's own detection magic to skip tests. If it ends up being wrong in some cases Cython tests will be silently skipped in CI.

Perhaps let's just check against the Python version and skip if it's Python 3.12?